netbox/core/views.py

```diff
@@ -642,11 +648,145 @@ def get(self, request):
+            'created': None,
+            'updated': None,
+            'is_local': True,
+            'is_installed': True,
+            'is_certified': False,
+            'is_community': False,
+            'versions': [],
+        }
+
+    return plugins
+
+
+def get_catalog_plugins(plugins):
+    url = 'https://api.netbox.oss.netboxlabs.com/v1/plugins'
+    session = requests.Session()
+
+    def get_pages():
+        payload = {'page': '1', 'per_page': '25'}
+        first_page = session.get(url, params=payload).json()
+        yield first_page
+        num_pages = first_page['metadata']['pagination']['last_page']
+
+        for page in range(2, num_pages + 1):
+            payload['page'] = page
+            next_page = session.get(url, params=payload).json()
+            yield next_page
+
+    for page in get_pages():
+        for data in page['data']:
+
+            versions = []
+            versions.append(data['release_latest'])
+            versions.extend(data['release_recent_history'])
+            if data['slug'] in plugins:
+                plugins[data['slug']]['is_local'] = False
+                plugins[data['slug']]['is_certified'] = data['release_latest']['is_certified']
+                plugins[data['slug']]['description_short'] = data['description_short']
+            else:
+                plugins[data['slug']] = {
+                    'slug': data['slug'],
+                    'config_name': data['config_name'],
+                    'name': data['title_short'],
+                    'title_long': data['title_long'],
+                    'tag_line': data['tag_line'],
+                    'description_short': data['description_short'],
+                    'author': data['author']['name'] or _('Unknown Author'),
+                    'created': datetime_from_timestamp(data['created_at']),
+                    'updated': datetime_from_timestamp(data['updated_at']),
+                    'is_local': False,
+                    'is_installed': False,
+                    'is_certified': data['release_latest']['is_certified'],
+                    'is_community': not data['release_latest']['is_certified'],
+                    'versions': versions,
+                }
+
+    return plugins
+
+
+def get_plugins():
+    if plugins := cache.get('plugins-catalog-feed'):
+        return plugins
+
+    plugins = {}
+    plugins = get_local_plugins(plugins)
+    plugins = get_catalog_plugins(plugins)
+
+    cache.set('plugins-catalog-feed', plugins, 3600)
+    return plugins
+
+
+class PluginListView(UserPassesTestMixin, View):
+
+    def test_func(self):
+        return self.request.user.is_staff
+
+    def get(self, request):
+        q = request.GET.get('q', None)
+
+        plugins = get_plugins()
+        if q:
+            plugins = [v for k, v in plugins.items() if q.casefold() in v['name'].casefold()]
+        else:
+            plugins = [v for k, v in plugins.items()]
```

Comment:
These can be simplified:
```suggestion
            plugins = plugins.values()
```